pkg/steps/pod.go

```diff
@@ -55,6 +55,16 @@ type PodStepConfiguration struct {
 	Clone              bool
 }
 
+type GeneratePodOptions struct {
+	IsObserver bool
+}
+
+func DefaultGeneratePodOptions() *GeneratePodOptions {
+	return &GeneratePodOptions{
+		IsObserver: false,
+	}
+}
+
```

Comment:
Isn't this functionally the same as passing `nil` to the functions?

Reply:
I see your point, and you are right. I moved it to `steps` because I need the information on whether the pod is an observer in `addPodUtils`. And this was the only structure that does that. If I had left it inside `multi_stage` I would have a cyclic dependency between those two packages. So what do you suggest? Create a new struct? I can't think of a good option.

Reply:
It's fine here, but the option should be about the effect it has on the generated pod.

- `steps/pod.go`: provides the option to propagate the error, knows nothing about observers
- `steps/multi_stage/multi_stage.go`: knows it is generating an observer and requests that option

Reply:
Now it is more clear to me, but I cannot just rename the option, because the `pkg/steps/multi_stage/gen.go` still uses the `IsObserver` in `isKubeconfigNeeded()` and `addSecretWrapper()`

Reply:
I'd still rather have flags as parameters than `bool`s, especially since there is one already (`clone`).  But it should present the options that the function provides:

```go
type GeneratePodOptions struct {
    Clone             bool
    PropagateExitCode bool
}
```

The other functions can then set these according to their own internal criteria (`isObserver`, etc.).

Reply:
Okay, but I still think the `IsObserver` param is needed there. 
`func isKubeconfigNeeded(step *api.LiteralTestStep, opts *generatePodOptions) bool {
	needsKubeconfig := step.NoKubeconfig == nil || !*step.NoKubeconfig
	return needsKubeconfig || opts.IsObserver
}` 
For example here, the kubeconfig should be added only if it is an observer, with only `Clone` and `PropagateExitCode` present, it would become unclear.

Reply:
It is needed, but it should not be in `pkg/steps`, since it's internal to `pkg/steps/multi_stage`.  We can translate from the private `generatePodOptions` to the public `GeneratePodOptions` inside the multi-stage module.

Reply:
That is a good idea! I hope I understood you correctly and addressed it with the latest changes.